integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestUpdateForPartitionTable.scala

```diff
@@ -0,0 +1,101 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.spark.testsuite.partition
+
+import org.apache.spark.sql.test.util.QueryTest
+import org.scalatest. BeforeAndAfterAll
+
+class TestUpdateForPartitionTable extends QueryTest with BeforeAndAfterAll {
+
+  override def beforeAll: Unit = {
+    dropTable
+
+    sql("create table test_range_partition_table (id int) partitioned by (name string) " +
+      "stored by 'carbondata' TBLPROPERTIES('PARTITION_TYPE' = 'RANGE','RANGE_INFO' = 'a,e,f')")
+    sql("create table test_hive_partition_table (id int) partitioned by (name string) " +
+      "stored by 'carbondata'")
+    sql("create table test_hash_partition_table (id int) partitioned by (name string) " +
+      "stored by 'carbondata' TBLPROPERTIES('PARTITION_TYPE' = 'HASH','NUM_PARTITIONS' = '2')")
+    sql("create table test_list_partition_table (id int) partitioned by (name string) " +
+      "stored by 'carbondata' TBLPROPERTIES('PARTITION_TYPE' = 'LIST','LIST_INFO' = 'a,e,f')")
+  }
+
+  def dropTable = {
+    sql("drop table if exists test_hash_partition_table")
+    sql("drop table if exists test_list_partition_table")
+    sql("drop table if exists test_range_partition_table")
+    sql("drop table if exists test_hive_partition_table")
+  }
+
+  ignore("test update for partition table") {
+    sql("drop table if exists test_partition_table")
+    sql("create table test_partition_table (id int) partitioned by (name string) " +
+      "stored by 'carbondata' TBLPROPERTIES('PARTITION_TYPE' = 'RANGE','RANGE_INFO' = 'a,e,f')")
+    sql("insert into test_partition_table select 1,'b' ")
+    sql("insert into test_partition_table select 2,'z' ")
+
+    sql("update test_partition_table set (name) = ('c') where id = 1").collect()
+    sql("update test_partition_table set (name) = ('d') where id = 1").collect()
+    sql("update test_partition_table set (name) = ('e') where id = 1").collect()
+    sql("update test_partition_table set (name) = ('f') where id = 1").collect()
+
+    // the result is wrong
+    sql("select * from test_partition_table").collect().foreach(println)
+    sql("drop table if exists test_partition_table")
+  }
+
+  test ("test update for range partition table") {
+    sql("insert into test_range_partition_table select 1,'b' ")
+
+    val ex = intercept[UnsupportedOperationException] {
+      sql("update test_range_partition_table set (name) = ('c') where id = 1").collect()
+    }
+
+    assertResult("Unsupported operation on range ,hash or list partition table")(ex.getMessage)
+  }
+
+  test ("test update for list partition table") {
```

Comment:
no need for the following two tests, there will be too much code if we go through all the possibilities OR you try to merge the three tests here using iteration to avoid too many code changes

Reply:
done